qt-models/divepicturemodel.cpp

```diff
@@ -56,25 +56,34 @@ void DivePictureModel::updateDivePictures()
 		endRemoveRows();
 	}
 
-	// if the dive_table is empty, ignore the displayed_dive
-	numberOfPictures = dive_table.nr == 0 ? 0 : dive_get_picture_count(&displayed_dive);
-	if (numberOfPictures == 0) {
+	// if the dive_table is empty, quit
+	if (dive_table.nr == 0)
 		return;
-	}
-
+	
 	stringPixmapCache.clear();
 	SPictureList pictures;
-	FOR_EACH_PICTURE_NON_PTR(displayed_dive) {
-		stringPixmapCache[QString(picture->filename)].offsetSeconds = picture->offset.seconds;
-		pictures.push_back(picture);
-	}
-
-	QList<SPixmap> list = QtConcurrent::blockingMapped(pictures, scaleImages);
-	Q_FOREACH (const SPixmap &pixmap, list)
-		stringPixmapCache[pixmap.first->filename].image = pixmap.second;
+	int i, numberOfPicturesOld;
+	struct dive *dive;
+	for_each_dive (i, dive) {
+		if (dive->selected) {
+			if (dive->id == displayed_dive.id)
+				rowDDStart = numberOfPictures;
+			numberOfPicturesOld = numberOfPictures;
+			numberOfPictures += dive_get_picture_count(dive);
+			if (dive->id == displayed_dive.id)
+				rowDDEnd = numberOfPictures;
+			FOR_EACH_PICTURE(dive) {
+				stringPixmapCache[QString(picture->filename)].offsetSeconds = picture->offset.seconds;
+				pictures.push_back(picture);
+			}
+		}
+		QList<SPixmap> list = QtConcurrent::blockingMapped(pictures, scaleImages);
```

Comment:
Note that you are repeating this for *every* dive. I think this code should go up one level...

Reply:
Yes, that's wrong. Thanks for pointing it out.